### PR TITLE
add a minimal `np.random` analog

### PR DESCRIPTION
**Pull request by ev-br**
Only mimic the old numpy random API.
No guarantees about reproducibility etc beyond those of pytorch.

For the moment, only add a minimal subset to help porting the numpy test suite.

Is on top of gh-22, for clarity.

### CODE REVIEW

File: torch_np/random.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+__all__ = ["seed", "random_sample", "sample", "random", "rand", "randn", "normal"]
+
+
+def seed(seed=None):
+    if seed is not None:
+        torch.random.manual_seed()
+
+
+def random_sample(size=None):
```
Comment (lezcano) [inline torch_np/random.py:26]: Why not simply calling this `random` or `sample`?

File: torch_np/random.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+
+sample = random_sample
+random = random_sample
+
+
+def uniform(low=0.0, high=1.0, size=None):
+    if size is None:
+        values = torch.rand(())
+        return float(low + (high - low) * values)
```
Comment (lezcano) [inline torch_np/random.py:46]: You may prefer to simply use `torch.uniform`.

Also, note that this formula overflows for `high = max_float, low=-max_float`. To account for this, you may want to divide high and low by 2 if they are too large and too small respectively.

File: torch_np/random.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+def rand(*size):
+    return random_sample(size)
+
+
+sample = random_sample
+random = random_sample
+
+
+def uniform(low=0.0, high=1.0, size=None):
```
Comment (lezcano) [inline torch_np/random.py:43]: You may want to write this function either using `torch.uniform` or doing `low + (high - low) * random(size)` (+ some proper handling of the overflow if you see fit).

File: torch_np/random.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        values = torch.rand(())
+        return float(low + (high - low) * values)
+    else:
+        values = torch.rand(size).to(_default_dtype)
+        return asarray(low + (high - low) * values)
+
+
+def randn(*size):
+    if size == ():
+        return float(torch.randn(size))
```
Comment (lezcano) [inline torch_np/random.py:54]: Can you encapsulate all this `size` logic into its own function?

File: torch_np/random.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    else:
+        values = torch.randn(*size).to(_default_dtypes)
+        return asarray(values)
+
+
+def normal(loc=0.0, scale=1.0, size=None):
+    if size is None:
+        size = ()
+    return loc + scale * randn(*size).to(_default_dtype)
```
Comment (lezcano) [inline torch_np/random.py:63]:
```suggestion
    return loc + scale * randn(*size)
```

**Review by lezcano — APPROVED** [review]
Just a few minor comments.

File: torch_np/random.py (hunk 5 above)
Comment (ev-br) [reply torch_np/random.py:63]: Why? Note that numpy's default is float64 not float32, and ISTM it's better to centralize this.

Comment (lezcano) [reply torch_np/random.py:63]: This logic of setting the default type and handling the size is already implemented in the `randn` function, so no need to repeat it here.

File: torch_np/random.py (hunk 1 above)
Comment (ev-br) [reply torch_np/random.py:33]: Numpy has these three aliases (random, random_sample and sample), all three are here.

Comment (lezcano) [reply torch_np/random.py:33]: ah, fair enough!

File: torch_np/random.py (hunk 3 above)
Comment (ev-br) [reply torch_np/random.py:48]: OK, refactored this to use `torch.empty(size).uniform_(...)`.
Am not going to even wonder why  `torch.rand` but  `torch.Tensor.uniform_` and no `torch.uniform`.

File: torch_np/random.py (hunk 5 above)
Comment (ev-br) [reply torch_np/random.py]: Ah, OK.  This is different in the last commit now, which implements 
`torch.empty(size, dtype=_default_dtype).uniform_(low, high)`, as requested.

File: torch_np/random.py (hunk 3 above)
Comment (lezcano) [reply torch_np/random.py:48]: large library, organically grown... jajajaj

**Pull request merged**